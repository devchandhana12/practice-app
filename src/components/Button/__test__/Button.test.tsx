// src/components/Button/__tests__/ButtonComp.test.js

import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ButtonComp from "../index";

describe("ButtonComp", () => {
  it("renders ButtonComp correctly", () => {
    const { asFragment } = render(
      <ButtonComp
        label="Click me"
        variant="primary"
        color="blue"
        style={{ fontWeight: "bold" }}
        type="button"
        disabled={false}
        onClick={() => console.log("Button clicked")}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("handles click event", () => {
    // Mocking a function to check if it's called when the button is clicked
    const mockClickHandler = jest.fn();

    const { getByText } = render(
      <ButtonComp label="Click me" onClick={mockClickHandler} />
    );

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const button = getByText("Click me");
    button.click();

    expect(mockClickHandler).toHaveBeenCalled();
  });
});
