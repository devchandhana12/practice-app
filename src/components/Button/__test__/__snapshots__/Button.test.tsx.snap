// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComp renders ButtonComp correctly 1`] = `
<DocumentFragment>
  <button
    class="fs-6 fw-bold fst-italic btn btn-primary"
    style="background-color: blue; font-weight: bold;"
    type="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
